include/nlohmann/ordered_map.hpp

```diff
@@ -96,31 +101,23 @@ template <class Key, class T, class IgnoredLess = std::less<Key>,
         return at(key);
     }
 
-    template<class KeyType, detail::enable_if_t<
-                 detail::is_usable_as_key_type<key_compare, key_type, KeyType>::value, int> = 0>
+    template<class KeyType, detail::enable_if_t<detail::is_usable_as_key_type<key_compare, key_type, KeyType>::value, int> = 0>
     const T & operator[](KeyType && key) const
     {
         return at(std::forward<KeyType>(key));
     }
 
     T& at(const key_type& key)
     {
-        for (auto it = this->begin(); it != this->end(); ++it)
-        {
-            if (m_compare(it->first, key))
-            {
-                return it->second;
-            }
-        }
-
-        JSON_THROW(std::out_of_range("key not found"));
+        const ordered_map<Key, T, IgnoredLess, Allocator>& cThis = *this;
+        return const_cast<T&>(cThis.at(key));
```

Comment:
Please don't do this. Casting away `const` is way more dangerous than the benefit of avoiding some code repetition.

Reply:
@nlohmann 
Thanks for correcting me.
I closed the PR.
I would like to know why the using of const cast is dangerous.
Thanks in advance